lib/msf/base/sessions/meterpreter.rb

```diff
@@ -199,6 +199,15 @@ def bootstrap(datastore = {}, handler = nil)
 
       # Tell the handler that we have a session
       handler.on_session(self) if handler
+
+      # Notify the framework that we have a new session opening up...
+      # Don't let errant event handlers kill our session
+      begin
+        framework.events.on_session_open(session)
+      rescue ::Exception => e
+        wlog("Exception in on_session_open event handler: #{e.class}: #{e}")
+        wlog("Call Stack\n#{e.backtrace.join("\n")}")
+      end
```

Comment:
I moved this block from inside a statement like `if session.register?`
`session.register?` only seems to return `false` for a  vncinject session, I don't fully understand why you don't register that kind of session but I didn't see any reason why you would't want to trigger this event on that kind of a session so I left it out, but I can add it back in if anyone has more insight